File: modules/jpa/testdata/java/reference/model/Employee.java
package com.example;

import javax.persistence.*;
import jakarta.persistence.*;
import java.util.List;

@Entity
@DiscriminatorValue("employee")
public class Employee extends BaseEntity {

    @Column
    private String name;

    @Column
    private String position;

    @ManyToOne
    @JoinColumn(name = "department_id")
    private Department department;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPosition() {
        return position;
    }

    public void setPosition(String position) {
        this.position = position;
    }

    public Department getDepartment() {
        return department;
    }

    public void setDepartment(Department department) {
        this.department = department;
    }
}

File: modules/jpa/testdata/kotlin/reference/model/Employee.kt
package com.example

import jakarta.persistence.*
import javax.persistence.*

@Entity
@DiscriminatorValue("employee")
class Employee : BaseEntity() {
    @Column
    var name: String? = null

    @Column
    var position: String? = null

    @ManyToOne
    @JoinColumn(name = "department_id")
    var department: Department? = null
}
